todotxtio.py

```diff
@@ -85,43 +114,57 @@ def from_string(string):
                 if todo_pre_data.group(2):
                     todo.completion_date = todo_pre_data.group(2)
             else:
-                todo.creation_date = todo_pre_data.group(2)
+                todo.creation_date = todo_pre_data.group(4)
```

Comment:
Well that's working ATM, maybe the regex is wrong. Better check the regex first. Anyway I never had issues with creation dates.

Reply:
please check with the following code (just open the Python interpreter and paste the code in. I did it for both, Python v2.7 and Python v3.6):

```python
import re
todo_data_regex = re.compile( \
    '^(?:(x) )?' + \
    '(?:(\d{4}-\d{2}-\d{2}) )?' + \
    '(?:\(([A-Z])\) )?' + \
    '(?:(\d{4}-\d{2}-\d{2}) )?' \
    )
match = todo_data_regex.match( \
    '(A) 2018-03-28 an unfinished task created at 2018-03-28 +TestProject @Home due:2018-03-29')
print( \
    'the complete prefix: ' + str(match.group(0)) + '\n' + \
    'the COMPLETED FLAG:  ' + str(match.group(1)) + '\n' + \
    'the COMPLETED DATE:  ' + str(match.group(2)) + '\n' + \
    'the PRIORITY:        ' + str(match.group(3)) + '\n' + \
    'the CREATION DATE:   ' + str(match.group(4)) \
    )
```

in both versions, the creation date should appear in ```match.group(4)``` . When inserting a COMPLETED TASK in the code above, the creation date appears in ```match.group(4)``` as well. As I understand RegEx there, the algorithm should not differ with a changed input string.